### PR TITLE
crimson: various fmt related changes and fixes

### PR DESCRIPTION
**Pull request by tchaikov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/object_data_handler.cc [hunk 1]
```diff
@@ -359,6 +359,13 @@ struct overwrite_plan_t {
     }
   }
 };
+} // namespace namespace crimson::os::seastore {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/object_data_handler.cc:362]: } // namespace crimson::os::seastore

File: src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage.h [hunk 2]
```diff
@@ -1669,27 +1669,30 @@ struct staged {
         }
       }
     }
-    std::ostream& print(std::ostream& os, bool is_top) const {
+
+    template<typename OutputIt>
+    auto do_format_to(OutputIt out, bool is_top) const {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage.h:1674]: need to drop the user of this method at lino 1733

File: src/crimson/os/seastore/object_data_handler.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/os/seastore/object_data_handler.cc:362]: done.

File: src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage.h (hunk 2 above)
Comment (tchaikov) [reply src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage.h:1674]: dropped.

**Comment by tchaikov** [comment]
@cyx1231st fixed and repushed. could you please take another look?

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Review by cyx1231st — APPROVED** [review]

**Pull request merged**